src/users/usersActions.js

```diff
@@ -25,17 +26,42 @@ export const reportPresence = (isActive: boolean = true, newUserInput: boolean =
   });
 };
 
-const typingWorker = auth => ({
-  get_current_time: () => new Date().getTime(),
+const typingWorker = (state: GlobalState) => {
+  const auth: Auth = getAuth(state);
 
-  notify_server_start: (user_ids_array: number[]) => {
-    api.typing(auth, JSON.stringify(user_ids_array), 'start');
-  },
+  // User ID arrays are only supported in server versions >= 2.0.0-rc1, for
+  // versions before this, email arrays are used. If current server version is
+  // undetermined, user ID arrays are optimistically used.
+  // Support for user ID arrays was added in zulip/zulip commit 2f634f8.
```

Comment:
nit: mention the commit ID right next to the version number -- they're two ways of describing the same fact